### ISSUE TITLE
ApiBehaviorApplicationModelProvider overwrites existing BindingInfo in entirety when inferring binding sources

### ISSUE THREAD

**Issue by pranavkm** [aspnet/Mvc#7609]
Problematic code: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs#L180-L186

More details:

`ApiBehaviorApplicationModelProvider` attempts to infer `BindingSource` for parameters that do not already have a `BindingSource` specified. Typically a `BindingSource` is calculated for a parameter, when it has attribute such as `FromQuery` or it's a special type that `ModelMetadata` knows about (such as `CancellationToken`, `IFormFile`) or has a `ModelBinderType` specified (e.g. `[ModelBinder(typeof(SomeBinderType))]`.  When a `BindingSource` is inferred, the provider overwrites the `BindingInfo` for the parameter, discarding other pieces of `BindingInfo` such as `BinderModelName`, `IPropertyFilterProvider`, and `IRequestPredicateProvider`. 

Here's a scenario that's broken:

```C#
[ApiController]
public class TestController
{
    [HttpGet("api")]
    public IActionResult ModelBinderAttribute([ModelBinder(Name = "top")] int value);
}
```

The resulting `BindingInfo` for `value` would have a `null` `BinderModelName`

**Comment by mkArtakMSFT** [comment]
Change `ModelBinder` to `FromQuery` as a workaround

```csharp
[ApiController]
public class TestController
{
    [HttpGet("api")]
    public IActionResult ModelBinderAttribute([FromQuery(Name = "top")] int value);
}
```

**Comment by pranavkm** [comment]
Looks like it affects more scenarios than I had originally accounted for. See https://github.com/aspnet/Mvc/issues/7799. More specifically, when specifying a custom binder type, the binder name is not passed through (it's always empty). You have to account for it in the binder which is ugly. Filed https://github.com/aspnet/Mvc/issues/7804 to port this to the patch release.

**Comment by johnkwaters** [comment]
Looks like it wasnt fixed in 2.1.1?
I tried removing
[FromForm(Name = "formfile")]
And it it doesn't work. I have to add this attribute, and the property that matches the variable name.
